### PR TITLE
HIVE-25003: Move iceberg-handler under a hive-iceberg module

### PR DESCRIPTION
**Pull request by pvary**
### What changes were proposed in this pull request?
Create a new module stucture:
* hive-iceberg
   * iceberg-handler
   * iceberg-catalog (later)

### Why are the changes needed?
Later we will have multiple iceberg modules and we need better structure

### Does this PR introduce _any_ user-facing change?
no

### How was this patch tested?
manual and unit tests

### CODE REVIEW

**Comment by pvary** [comment]
cc: @marton-bod

File: iceberg/pom.xml [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+                        <goals>
+                            <goal>check</goal>
+                        </goals>
+                    </execution>
+                </executions>
+            </plugin>
+        </plugins>
+    </build>
+
+<!--    <profiles>-->
```
Comment (marton-bod) [inline iceberg/pom.xml:311]: Isn't it enough to keep this either here or just in the child modules?

File: iceberg/patched-iceberg-core/pom.xml [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+                        <configuration>
+                            <artifactItems>
+                                <artifactItem>
+                                    <groupId>org.apache.iceberg</groupId>
+                                    <artifactId>iceberg-core</artifactId>
+                                    <version>${iceberg-api.version}</version>
+                                    <type>jar</type>
+                                    <overWrite>true</overWrite>
+                                    <outputDirectory>${project.build.directory}/classes</outputDirectory>
+                                    <excludes>
```
Comment (marton-bod) [inline iceberg/patched-iceberg-core/pom.xml:69]: I think I'm missing some step here: don't we need to list the SnapshotSummary class to be replaced here?

File: iceberg/pom.xml (hunk 1 above)
Comment (pvary) [reply iceberg/pom.xml:311]: Might worth to check if we want to keep it.

We should spend some time on the errorprone plugin and decide what to do with it. I was not able to compile the code reliably in IntelliJ with the plugin enabled.

File: iceberg/patched-iceberg-core/pom.xml (hunk 2 above)
Comment (pvary) [reply iceberg/patched-iceberg-core/pom.xml:69]: Those will come with the individual changes

File: pom.xml [hunk 3]
```diff
@@ -64,8 +64,7 @@
     <module>standalone-metastore</module>
     <module>upgrade-acid</module>
     <module>kafka-handler</module>
-    <module>iceberg-handler</module>
-    <module>iceberg-catalog</module>
+    <module>iceberg</module>
```
Comment (marton-bod) [inline pom.xml:67]: Just to check, after this refactor, the handler and catalog jars are still packaged into the `packaging/target/.../lib` directory? How about the patched jars (core and api)? Are they correctly included into the shaded handler jar with the patched classfiles, but the patches jars are not included into the packaging `/lib`?

Comment (pvary) [reply pom.xml:67]: That's a good question!
I will check...

File: iceberg/pom.xml (hunk 1 above)
Comment (marton-bod) [reply iceberg/pom.xml:311]: As long as the upstream Iceberg community keeps the baseline plugins, I think we should have it too. But I agree it's painful with IntelliJ.

Perhaps we can hide it behind a profile flag and run it only on demand? I wonder if there's some profile/property the CI builds use which we could use to activate this only during CI runs. That would mean we'd get to run it locally only on demand (therefore not messing up the IDE) but still enforce the validation checks as a precommit requirement.

Comment (marton-bod) [reply iceberg/pom.xml:311]: If there's nothing better, we could perhaps use the `dist` profile and run the errorprone checks only then

File: pom.xml (hunk 3 above)
Comment (pvary) [reply pom.xml:67]: Applied my other patches so there is a new class in `patched-iceberg-api` (`CommitStateUnknownException`)
Run the packaging and checked the `packaging/target/.../lib` and it still contains the libs:
```
$ cd packaging/target/apache-hive-4.0.0-SNAPSHOT-bin/apache-hive-4.0.0-SNAPSHOT-bin/
$ ls *iceberg*
hive-iceberg-catalog-4.0.0-SNAPSHOT.jar	hive-iceberg-handler-4.0.0-SNAPSHOT.jar
```

The new class is in the jar:
```
$ zip -sf hive-iceberg-handler-4.0.0-SNAPSHOT.jar |grep CommitStateUnknownException
  org/apache/hive/iceberg/org/apache/iceberg/exceptions/CommitStateUnknownException.class
```

So this seems ok

Comment (marton-bod) [reply pom.xml:67]: Great, thanks

File: iceberg/pom.xml (hunk 1 above)
Comment (pvary) [reply iceberg/pom.xml]: Used a profile to handle `errorProne`.

See http://ci.hive.apache.org/blue/organizations/jenkins/hive-precommit/detail/PR-2174/1/pipeline/

Comment (marton-bod) [reply iceberg/pom.xml]: Perfect!

File: iceberg/pom.xml [hunk 4]
```diff
@@ -0,0 +1,325 @@
...
+            </dependency>
+
+            <dependency>
+                <groupId>org.apache.avro</groupId>
+                <artifactId>avro</artifactId>
+                <version>${iceberg.avro.version}</version>
+            </dependency>
+            <dependency>
+                <groupId>org.apache.orc</groupId>
+                <artifactId>orc-core</artifactId>
```
Comment (marton-bod) [inline iceberg/pom.xml:192]: I think we're getting this via `iceberg-core` already, no? I don't think it had to be declared as a dependency for handler or catalog modules

File: iceberg/pom.xml [hunk 5]
```diff
@@ -0,0 +1,325 @@
...
+                        <groupId>org.apache.avro</groupId>
+                        <artifactId>avro</artifactId>
+                    </exclusion>
+                </exclusions>
+            </dependency>
+
+            <!-- test dependencies -->
+            <dependency>
+                <groupId>org.apache.hive</groupId>
+                <artifactId>hive-service</artifactId>
```
Comment (marton-bod) [inline iceberg/pom.xml:163]: hive-service and avro are not just test dependencies

File: iceberg/pom.xml [hunk 6]
```diff
@@ -0,0 +1,325 @@
...
+                </exclusions>
+            </dependency>
+            <dependency>
+                <groupId>org.apache.hive</groupId>
+                <artifactId>hive-serde</artifactId>
+                <version>${project.version}</version>
+            </dependency>
+            <dependency>
+                <groupId>org.apache.hive</groupId>
+                <artifactId>hive-standalone-metastore-server</artifactId>
```
Comment (marton-bod) [inline iceberg/pom.xml:139]: I think we only used the `metastore-common` dependency - do we need to declare the server as well? it's listed below with the `tests` classifier already

File: iceberg/pom.xml [hunk 7]
```diff
@@ -0,0 +1,325 @@
...
+            <dependency>
+                <groupId>org.apache.hive</groupId>
+                <artifactId>hive-iceberg-catalog</artifactId>
+                <classifier>tests</classifier>
+                <version>${project.version}</version>
+            </dependency>
+
+            <dependency>
+                <groupId>org.apache.avro</groupId>
+                <artifactId>avro</artifactId>
```
Comment (marton-bod) [inline iceberg/pom.xml:187]: I think avro is not just a test dependency - can move this above the comment?

File: iceberg/pom.xml (hunk 6 above)
Comment (pvary) [reply iceberg/pom.xml]: removed

File: iceberg/pom.xml (hunk 7 above)
Comment (pvary) [reply iceberg/pom.xml]: removed

File: iceberg/pom.xml (hunk 4 above)
Comment (pvary) [reply iceberg/pom.xml]: removed

**Review by marton-bod — APPROVED** [review]

**Review by lcspinter — APPROVED** [review]

**Pull request merged**